regression/ansi-c/enum9/main.c

```diff
@@ -0,0 +1,20 @@
+typedef enum : unsigned
+{
+  X
+} my_enum1;
+
+enum my_enum2 : unsigned
+{
+  Y
+};
+
+struct S
+{
+  enum my_enum2 : unsigned a;
+  enum my_enum2 : unsigned b : 2;
+};
+
+int main()
+{
+  enum my_enum2 : unsigned enum_var1;
```

Comment:
Huh? Any ideas on what this means? Seems like clang allows this notation, but it appears to not actually mean anything, for instance

```
#include <stdio.h>

enum my_enum : signed char {
  A, B
};


int main(void) {
  enum my_enum : unsigned long long  X = 255;
  printf("%d\n", X);
}
```

Reply:
Yeah it seems to just be ignored by clang. My guess would be that it might have been overly complicated for clang to exclude this case in the grammar, which would also be true for cbmc. I think we should report this as an error during conversion/typechecking.